### ISSUE TITLE
Number.is_nan is too slow

### ISSUE THREAD

**Issue by JaroslavTulach** [enso-org/enso#5712]
### What type of issue is this?

Permanent – Ocurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

I tried to verify effect of [performance bugfix](https://github.com/enso-org/enso/commit/58c7ca54011bba1208de298d7b2e6877570d5b47) on performance of the [Sieve of Eratosthenes](https://github.com/jtulach/sieve/blob/5ff0f8881aed3d0b772c532e22665361ad8f0854/enso/sieve.enso). It is still 10x slower then it should be.

![Number.is_nan](https://user-images.githubusercontent.com/26887752/220303493-e2bcc326-9f49-4f06-89f5-17006f5162d9.png)

According to the _VisualVM Polyglot Sampler_ it looks like `Numbers.is_nan` performance is too slow as 22% of the time is spent there.

### Expected behaviour

`Number.is_nan` has to be almost no-op.

### Enso Version

58c7ca54011bba1208de298d7b2e6877570d5b47

### Browser or standalone distribution

Standalone distribution

### Browser Version or standalone distribution

standalone

### Operating System

Linux

### Operating System Version

Ubuntu

### Hardware you are using

AMD Radeon

**Comment by Akirathan** [comment]
@JaroslavTulach Thanks for the report, will look into a workaround for `Number.is_nan`.

**Comment by Akirathan** [comment]
After some investigation, I realized that the performance problem is not in `Number.is_nan` method itself, but rather in `Comparable.from (_:Number)` that was implemented as:
```
Comparable.from (that:Number) =
    case that.is_nan of
        True -> Incomparable
        False -> Default_Ordered_Comparator
```
Which is a violation of the API - there should be a binary mapping between a `Type` and its Comparator. Ideally, we want to have `NaN` as a separate type (#5739) and cache the Comparator<->Type mapping (#5740)

**Comment by Akirathan** [comment]
Solved with #5744
